File: ProgramaAssistenciaTecnicaPython/OrdemServico.py
from Tecnico import *
from NotaFiscal import *
from Aparelho import *

class OrdemServico:
    def __init__(self, nomeDono, email, tipo, problema, nomeTecnico, registro, feito, valorServico, pagamento):
        self.registro = registro
        self.infAparelho = Aparelho(tipo, problema, nomeDono, email)
        self.infTecnico = Tecnico(nomeTecnico)
        self.manutencao = Tecnico(nomeTecnico).realizarManutencao(feito)
        self.valorManutencao = Tecnico(nomeTecnico).custearServico(valorServico)
        self.pagou = Aparelho(tipo, problema, nomeDono, email).dono.realizarPagamento(pagamento)

    def mostraInformacoes(self):
        print(f"Nome do Cliente:  {self.infAparelho.dono.nome}")
        print(f"Email: {self.infAparelho.dono.email}")
        print(f"Tipo do aparelho: {self.infAparelho.tipo}")
        print(f"Problema: {self.infAparelho.problema}")
        print(f"Técnico responsável: {self.infTecnico.nome} ")
        print(f"Registro: {self.registro}")
        print(f"Manutencao: {self.manutencao}")
        print(f"Valor da manutencao: R$ {self.valorManutencao:,.2f}")
        nota = NotaFiscal(self.valorManutencao)
        nota.verificaPagamento(self.pagou)

File: aula02/pw-my-assist/src/OrdemServico.java
public class OrdemServico {
    private int registro;
    private Aparelho infAparelho;
    private Tecnico infTecnico;
    private boolean manutencao;
    private double valorManutencao;
    private boolean pagou;

    public OrdemServico(String nomeDono, String email, String tipo, String problema, String nomeTecnico, int registro, boolean feito, double valorServico, boolean pagamento) {
        this.setRegistro(registro);
        this.infAparelho = new Aparelho(tipo, problema, nomeDono, email);
        this.infTecnico = new Tecnico(nomeTecnico);
        this.manutencao = feito;
        this.valorManutencao = new Tecnico(nomeTecnico).custearServico(valorServico);
        this.pagou = new Aparelho(tipo, problema, nomeDono, email).getDono().realizarPagamento(pagamento);
    }

    public int getRegistro() {
        return registro;
    }

    public void setRegistro(int registro) {
        this.registro = registro;
    }

    public Aparelho getInfAparelho() {
        return infAparelho;
    }

    public void setInfAparelho(Aparelho infAparelho) {
        this.infAparelho = infAparelho;
    }

    public Tecnico getInfTecnico() {
        return infTecnico;
    }

    public void setInfTecnico(Tecnico infTecnico) {
        this.infTecnico = infTecnico;
    }

    public boolean isManutencao() {
        return manutencao;
    }

    public void setManutencao(boolean manutencao) {
        this.manutencao = manutencao;
    }

    public double getValorManutencao() {
        return valorManutencao;
    }

    public void setValorManutencao(double valorManutencao) {
        this.valorManutencao = valorManutencao;
    }

    public boolean isPagou() {
        return pagou;
    }

    public void setPagou(boolean pagou) {
        this.pagou = pagou;
    }

    public void mostraInformacoes() {
        System.out.println("Nome do Cliente:  " + this.infAparelho.getDono().getNome());
        System.out.println("Email: " + this.infAparelho.getDono().getEmail());
        System.out.println("Tipo do aparelho: " + this.infAparelho.getTipo());
        System.out.println("Problema: " + this.infAparelho.getProblema());
        System.out.println("Técnico responsável: " + this.infTecnico.getNome());
        System.out.println("Registro: " + this.registro);
        System.out.println("Manutencao: " + this.manutencao);
        System.out.println("Valor da manutencao: R$ " + String.format("%.2f", this.valorManutencao));
        NotaFiscal nota = new NotaFiscal(this.valorManutencao);
        nota.verificaPagamento(this.pagou);
    }
}
